tensorboard/webapp/widgets/histogram/histogram_v2_test.ts

```diff
@@ -155,14 +175,77 @@ describe('histogram v2 test', () => {
       fixture.componentInstance.mode = HistogramMode.OFFSET;
       fixture.componentInstance.timeProperty = TimeProperty.STEP;
       fixture.detectChanges();
+      intersectionObserver.simulateVisibilityChange(fixture, true);
 
       expect(
         getAxisLabelText(fixture.debugElement.query(byCss.X_AXIS))
       ).toEqual(['-100', '0', '100']);
     });
+
+    it('updates axis only when chart is visible', () => {
+      const fixture = createComponent('foo', [
+        buildHistogramDatum({
+          step: 0,
+          bins: [buildBin({x: 1, dx: 5})],
+        }),
+        buildHistogramDatum({
+          step: 1,
+          bins: [buildBin({x: 0, dx: 100})],
+        }),
+        buildHistogramDatum({
+          step: 100,
+          bins: [buildBin({x: -100, dx: 5})],
+        }),
+      ]);
+      fixture.componentInstance.mode = HistogramMode.OFFSET;
+      fixture.componentInstance.timeProperty = TimeProperty.STEP;
+      fixture.detectChanges();
+      intersectionObserver.simulateVisibilityChange(fixture, true);
+      const beforeAxisLabels = getAxisLabelText(
+        fixture.debugElement.query(byCss.X_AXIS)
+      );
+      fixture.detectChanges();
+
+      intersectionObserver.simulateVisibilityChange(fixture, false);
+      fixture.detectChanges();
+
+      fixture.componentInstance.data = [
+        buildHistogramDatum({
+          step: 0,
+          bins: [buildBin({x: 1, dx: 5})],
+        }),
+        buildHistogramDatum({
+          step: 1,
+          bins: [buildBin({x: 0, dx: 100})],
+        }),
+      ];
+      fixture.detectChanges();
+      expect(
+        getAxisLabelText(fixture.debugElement.query(byCss.X_AXIS))
+      ).toEqual(beforeAxisLabels);
+
+      intersectionObserver.simulateVisibilityChange(fixture, true);
+      fixture.detectChanges();
+
+      expect(
+        getAxisLabelText(fixture.debugElement.query(byCss.X_AXIS))
+      ).toEqual(['0', '50', '100']);
+    });
   });
 
   describe('y axis render', () => {
+    it('does not render until component becomes visible', () => {
```

Comment:
This test case is named similarly to the one on L143 for the x axis.  However, the x axis one asserts the axis label text is empty before changing visibility to false, while this case does not.  Should they be consistent?